charts/spire/charts/spire-lib/templates/spire-server/_controller-manager-roles.yaml

```diff
@@ -0,0 +1,100 @@
+{{/* Takes in a dictionary with keys:
+ * fullName - the full name of the controller manager
+ * releaseNamespace - the namespace the release was made into
+ * namespace - the namespace the server is installed into
+ * subject - subject of the role bindings
+ * controllerManagerEnabled - enabled setting for the controller manager
+ * validatingWebhookConfigurationEnabled - enabled setting for the validating webhook configuration
+*/}}
+{{- define "spire-lib.controller-manager-roles" }}
```

Comment:
Also see below, doesn't really seem to be a utility function that belongs in a library. 

Seems more like something that belongs in a spire-controller-manager chart. without the need of adding all kind of macro logic and complexity with dictionaries. 

Feels like we are completly going overboard with refactoring into a non-deterministic way of writing Helm charts that allow community to easily contribute and understand this chart.